src/test/java/com/spring/rest/dao/impl/CarCharacteristicsDaoImplTest.java

```diff
@@ -0,0 +1,97 @@
+package com.spring.rest.dao.impl;
+
+import com.spring.rest.dao.CarCharacteristicsDao;
+import com.spring.rest.model.CarCharacteristics;
+import com.spring.rest.model.CarMake;
+import com.spring.rest.model.CarModel;
+import com.spring.rest.model.FuelType;
+import com.spring.rest.model.Transmission;
+import org.junit.jupiter.api.Test;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.dao.DataIntegrityViolationException;
+import org.springframework.dao.EmptyResultDataAccessException;
+import java.util.Date;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+
+@SpringTestConfiguration
+class CarCharacteristicsDaoImplTest {
+
+    CarCharacteristicsDao carCharacteristicsDao;
+
+    @Autowired
+    CarCharacteristicsDaoImplTest(CarCharacteristicsDao carCharacteristicsDao) {
+        this.carCharacteristicsDao = carCharacteristicsDao;
+    }
+
+    @Test
+    void getCarsCharacteristics() {
+        assertEquals(1, carCharacteristicsDao.getCarCharacteristics(1).getCarId());
+        carCharacteristicsDao.deleteCarCharacteristics(5);
+        assertEquals(4, carCharacteristicsDao.getCarsCharacteristics().size());
+    }
+
+    @Test
+    void testGetNonExistentCarsCharacteristics() {
+        assertThrows(EmptyResultDataAccessException.class, () -> carCharacteristicsDao.getCarCharacteristics(30));
+    }
+
+    @Test
+    void getCarCharacteristics() {
+        assertEquals(5, carCharacteristicsDao.getCarsCharacteristics().size());
+    }
+
+    @Test
+    void addCarCharacteristics() {
+        int size = carCharacteristicsDao.getCarsCharacteristics().size();
+        int index = carCharacteristicsDao.addCarCharacteristics(new CarCharacteristics(2, new Date(32000),
+                133455, new FuelType(2), new Transmission(1), new CarModel(3,
+                new CarMake(2))));
+        assertEquals(size + 1, carCharacteristicsDao.getCarsCharacteristics().size());
+        assertEquals(133455, carCharacteristicsDao.getCarCharacteristics(index).getMileage());
```

Comment:
Get saved object just once and check all fields. In your approach you call DB a few times. What for?